
### Kinematic configuration parameter

# Debug mode
activate_output: true
activate_debug_output: false
activate_controller_node_output: true
plotting_result: true
publish_feedback: true

# Clock frequency //hz
clock_frequency: 20

#Optimal control problem dimensions
state_dim: 6
control_dim: 4

#Inputs and output topics topic
output_cmd: '/cmd_vel'   #'/jackal_velocity_controller/cmd_vel'
robot_state_topic: '/robot_state'

# Kinematic chain parameter
robot_base_link: base_link
target_frame: map
tracking_frame: map

# Weight on the slack variable for collision collision
slack_weight: 10000
repulsive_weight: 0.2
reference_velocity: 0.5

# Parameters for predefined reference path

## Path for cyberzoo_twobox environment
#global_path:
#      x: [0, 1.5, 3, 4.5, 3, 1.5, 0, -1.5, 0]
#      y: [0, 2.5, 5, 3, 0, 2.5, 5, 2.5, 0]
#      theta: [0, 1.5708, 0, -1.5708, 3.1416, 1.5708, 3.1416, -1.5708, 0]
#      n_poly_per_clothoid: 2

#global_path:
#      x: [0, 2, 4, 6, 6, 6, 4, 2, 0]
#      y: [0, 0, 0, 2, 4, 6, 8, 8, 8]
#      theta: [0, 0, 0, 1.5708, 1.5708, 1.5708, 3.1416, 3.1416, 3.1416]
#      n_poly_per_clothoid: 2

global_path:
      x: [0, 2, 4, 6, 8, 10, 12, 14, 16]
      y: [0, 0, 0, 0, 0, 0, 0, 0, 0]
      theta: [0, 0, 0, 0, 0, 0, 0, 0, 0]
      n_poly_per_clothoid: 2

# Parameters for collision constraints
obstacles:
      n_obstacles: 2
      sub_ellipse_topic: ellipse_objects_feed
      n_discs: 1
      ego_l: 0.5
      ego_w: 0.4

constraints:
     velocity_constraints:
           min: [-1, -1]
           max: [1,1]
     #orientation_constraints:

acado_config:
     max_num_iteration: 70
     kkt_tolerance: 1e-3
     integrator_tolerance: 1e-3
     start_time_horizon: 0.0
     end_time_horizon: 5.0 # 5.0  # control and prediction horizon
     discretization_intervals: 50
     sampling_time: 0.1
     weight_factors:
           contour_weight_factors:
                #always 6 component 3 linear/position and 3 angular/Euler angle
                [1.0, 1.0]
           control_weight_factors:
                # same as degree of freedom
                [1.5, 0.1]

